shared/cards.ts

```diff
@@ -3,35 +3,40 @@ export const VOTE_COFFEE = 'coffee';
 export const VOTE_OBSERVER = 'observer';
 export const VOTE_NOTE_VOTED = 'not-voted';
 
-export type CardValue =
-  | typeof VOTE_OBSERVER
-  | typeof VOTE_NOTE_VOTED
-  | typeof VOTE_COFFEE
-  | '?'
-  | '∞'
-  | 'XS'
-  | 'S'
-  | 'M'
-  | 'L'
-  | 'XL'
-  | 'XXL'
-  | '0'
-  | '0.5'
-  | '1'
-  | '2'
-  | '3'
-  | '4'
-  | '5'
-  | '8'
-  | '13'
-  | '16'
-  | '20'
-  | '21'
-  | '32'
-  | '34'
-  | '40'
-  | '55'
-  | '64'
-  | '89'
-  | '100'
-  | '128';
+export const SPECIAL_VALUES_ORDERED = ['∞', '?', VOTE_COFFEE] as const;
+
+const ABSTAINING_VOTES_ORDERED = [VOTE_NOTE_VOTED, VOTE_OBSERVER] as const;
+
+export const SIZES_ORDERED = ['XS', 'S', 'M', 'L', 'XL', 'XXL'] as const;
+
+const NUMERIC_VALUES_ORDERED = [
+  '0',
+  '0.5',
+  '1',
+  '2',
+  '3',
+  '4',
+  '5',
+  '8',
+  '13',
+  '16',
+  '20',
+  '21',
+  '32',
+  '34',
+  '40',
+  '55',
+  '64',
+  '89',
+  '100',
+  '128',
+] as const;
+
+export const ALL_VALUES_ORDERED = [
+  ...NUMERIC_VALUES_ORDERED,
+  ...SIZES_ORDERED,
+  ...SPECIAL_VALUES_ORDERED,
+  ...ABSTAINING_VOTES_ORDERED,
+] as const;
```

Comment:
While we do not have a performance problem, the current sorting has complexity O(n^2*log(n)) with the four `indexOf` lookups in the sorting. For a large number of values on a very weak device this could start to have effects. This could be addressed by building a reverse-lookup Map in cards.ts.

This could be as simple as

```js
const CARD_ORDER_BY_VALUE = new Map(ALL_VALUES_ORDERED.map((value, index) => [value, index]));
```

then you can replace

```js
ALL_VALUES_ORDERED.indexOf(value)
```

with

```js
CARD_ORDER_BY_VALUE .get(value)
```

Reply:
Implemented your comment. I had to deal with `undefined`s in `compareCardValues` now (although theoretically there should not be any `undefined`s), hope that works for you. :D